lib/min_heap.rb

```diff
@@ -39,33 +45,71 @@ def to_s
     end
 
     output += @store.last.value + "]"
-      
+
     return output
   end
 
   # This method returns true if the heap is empty
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity:  O(1)
+  # Space complexity: O(1)
   def empty?
-    raise NotImplementedError, "Method not implemented yet..."
+    return @store[0].nil?
   end
 
   private
 
   # This helper method takes an index and
   #  moves it up the heap, if it is less than it's parent node.
   #  It could be **very** helpful for the add method.
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity:  O(log n)
+  # Space complexity: O(log n)
   def heap_up(index)
-    
+    parent_index = get_parent(index)
+    if @store[parent_index].key > @store[index].key
+      swap(parent_index, index)
+      if parent_index > 0
+        heap_up(parent_index)
+      end
+    end
+    return
   end
 
-  # This helper method takes an index and 
+  # This helper method takes an index and
   #  moves it up the heap if it's smaller
   #  than it's parent node.
   def heap_down(index)
```

Comment:
Nice work, this does the job.  I like the helper methods you used here.  Well done!